eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/client/EclairRpcClient.scala

```diff
@@ -0,0 +1,382 @@
+  }
+
+  // TODO: channelstats, audit, networkfees?
+  // TODO: Add types
+
+  private def eclairCall[T](
+    command: String,
+    parameters: List[JsValue] = List.empty)(
+    implicit
+    reader: Reads[T]): Future[T] = {
+    val request = buildRequest(getDaemon, command, JsArray(parameters))
+    val responseF = sendRequest(request)
+
+    val payloadF: Future[JsValue] = responseF.flatMap(getPayload)
+
+    payloadF.map { payload =>
+      parseResult((payload \ resultKey).validate[T], payload)
+    }
+  }
+
+  case class RpcError(code: Int, message: String)
+  implicit val rpcErrorReads: Reads[RpcError] = Json.reads[RpcError]
+
+  private def parseResult[T](result: JsResult[T], json: JsValue): T = {
+    result match {
+      case res: JsSuccess[T] => res.value
+      case res: JsError =>
+        (json \ errorKey).validate[RpcError] match {
+          case err: JsSuccess[RpcError] =>
+            logger.error(s"Error ${err.value.code}: ${err.value.message}")
+            throw new RuntimeException(
+              s"Error ${err.value.code}: ${err.value.message}")
+          case _: JsError =>
+            logger.error(JsError.toJson(res).toString())
+            throw new IllegalArgumentException(
+              s"Could not parse JsResult: ${(json \ resultKey).get}")
+        }
+    }
+  }
+
+  private def getPayload(response: HttpResponse): Future[JsValue] = {
+    val payloadF = response.entity.dataBytes.runFold(ByteString.empty)(_ ++ _)
+
+    payloadF.map { payload =>
+      Json.parse(payload.decodeString(ByteString.UTF_8))
+    }
+  }
+
+  def sendRequest(req: HttpRequest): Future[HttpResponse] = {
+    Http(m.system).singleRequest(req)
+  }
+
+  def buildRequest(instance: EclairInstance, methodName: String, params: JsArray): HttpRequest = {
+    val uuid = UUID.randomUUID().toString
+
+    val obj: JsObject = JsObject(
+      Map(
+        "method" -> JsString(methodName),
+        "params" -> params,
+        "id" -> JsString(uuid)))
+
+    val uri = "http://" + instance.rpcUri.getHost + ":" + instance.rpcUri.getPort
+    val username = instance.authCredentials.username
+    val password = instance.authCredentials.password
+    HttpRequest(
+      method = HttpMethods.POST,
+      uri,
+      entity =
+        HttpEntity(ContentTypes.`application/json`, obj.toString))
+      .addCredentials(
+        HttpCredentials.createBasicHttpCredentials(username, password))
+  }
+
+  // TODO: THIS IS ALL HACKY
+
+  private def pathToEclairJar: String = {
+    System.getenv("ECLAIR_PATH") + "/eclair-node-0.2-beta5-8aa51f4.jar"
+  }
+
+  private var process: Option[Process] = None
+
+  def start(): Unit = {
+    val p = Process("java -jar -Declair.datadir=" + instance.authCredentials.datadir + s" ${pathToEclairJar} &")
+    val result = p.run()
+    logger.info(s"Starting eclair with datadir ${instance.authCredentials.datadir}")
+
+    process = Some(result)
+    ()
+  }
+
+  def isStarted(): Boolean = {
```

Comment:
We could probably do a check for `process == None` before stalling the program that calls this for 1.second